tokio-threadpool/src/pool/mod.rs

```diff
@@ -525,22 +525,20 @@ impl Pool {
         thread_local!(static THREAD_RNG_KEY: UnsafeCell<Option<XorShiftRng>> = UnsafeCell::new(None));
 
         THREAD_RNG_KEY.with(|t| {
-            #[cfg(target_pointer_width = "32")]
             fn new_rng(thread_id: usize) -> XorShiftRng {
-                XorShiftRng::from_seed([
-                    thread_id as u32,
-                    0x00000000,
-                    0xa8a7d469,
-                    0x97830e05])
-            }
-
-            #[cfg(target_pointer_width = "64")]
-            fn new_rng(thread_id: usize) -> XorShiftRng {
-                XorShiftRng::from_seed([
+                let seed = [
                     thread_id as u32,
                     (thread_id >> 32) as u32,
                     0xa8a7d469,
-                    0x97830e05])
+                    0x97830e05
+                ];
+
+                fn u32_as_u8<'a>(src: [u32; 4]) -> [u8; 16] {
+                    use std::mem::transmute;
+                    unsafe { transmute(src) }
```

Comment:
I am so sorry for unsafe but I could not manage to do it without this hack.

See https://docs.rs/rand/0.5.3/rand/prng/struct.XorShiftRng.html#method.from_seed , `type Seed = [u8; 16]`